sites/partners/src/listings/PaperListingForm/index.tsx

```diff
@@ -456,54 +442,87 @@ const ListingForm = ({ listing, editMode }: ListingFormProps) => {
   }, [listing, setUnits, setOpenHouseEvents])
 
   // eslint-disable-next-line @typescript-eslint/unbound-method
-  const { handleSubmit, getValues } = formMethods
+  const { handleSubmit, getValues, setError, clearErrors, reset } = formMethods
+
+  const triggerSubmitWithStatus = (status: ListingStatus) => {
+    triggerSubmit({ ...getValues(), status })
+  }
 
   const triggerSubmit = (data: FormListing) => {
     setAlert(null)
-    setLoading(true)
-    setSubmitData({ ready: true, data: { ...submitData.data, ...data } })
+    void onSubmit({ ...defaultValues, ...data })
   }
 
   /*
     @data: form data comes from the react-hook-form
   */
   const onSubmit = useCallback(
-    async (data: FormListing, status: ListingStatus) => {
-      try {
-        data = {
-          ...data,
-          status,
-        }
-        const orderedPreferences = preferences.map((pref, index) => {
-          return { ...pref, ordinal: index + 1 }
-        })
-        const formattedData = formatFormData(
-          data,
-          units,
-          openHouseEvents,
-          orderedPreferences,
-          latLong,
-          customMapPositionChosen
-        )
-        const result = editMode
-          ? await listingsService.update({
-              listingId: listing.id,
-              body: { id: listing.id, ...formattedData },
-            })
-          : await listingsService.create({ body: formattedData })
-        setLoading(false)
-        if (result) {
-          setSiteAlertMessage(
-            editMode ? t("listings.listingUpdated") : t("listings.listingSubmitted"),
-            "success"
+    async (formData: FormListing) => {
+      if (!loading) {
+        try {
+          setLoading(true)
+          clearErrors()
+          const orderedPreferences = preferences.map((pref, index) => {
+            return { ...pref, ordinal: index + 1 }
+          })
+          const formattedData = formatFormData(
+            formData,
+            units,
+            openHouseEvents,
+            orderedPreferences,
+            latLong,
+            customMapPositionChosen
           )
-
-          await router.push(`/listings/${result.id}`)
+          const removeEmptyFields = (obj) => {
+            Object.keys(obj).forEach(function (key) {
+              if (obj[key] && typeof obj[key] === "object") {
+                removeEmptyFields(obj[key])
+              }
+              if (obj[key] === null || obj[key] === undefined || obj[key] === "") {
+                delete obj[key]
+              }
+              if (
+                typeof obj[key] === "object" &&
+                !Array.isArray(obj[key]) &&
+                Object.keys(obj[key]).length === 0
+              ) {
+                delete obj[key]
+              }
+            })
+          }
+          removeEmptyFields(formattedData)
```

Comment:
Looks like a good candidate to be a helper function in a separate file 👀 Especially that this component is veeeeery long.

Reply:
Good idea :)